packages/director/src/screenshots/s3/s3.ts

```diff
@@ -49,7 +50,7 @@ export const getUploadUrl = async ({
         }
         return resolve({
           readUrl: `${S3_READ_URL_PREFIX || BUCKET_URL}/${key}`,
-          uploadUrl,
+          uploadUrl: S3_UPLOAD_URL || uploadUrl,
```

Comment:
@swandogg that would use the same URL for every artifact. Is that the desired behavior?

Reply:
hmm, not exactly I guess, in our use case we have a different base URL... say we wanted to upload to `s3api-core-example.com`, would this not work for that use case? would we have to add some string interpolation for other possibly necessary keys?

Reply:
In general, I assume something like `<baseURL>/{$key}` would work for most cases. 

---

The URL generated by AWS S3 looks like this:

```
https://presignedurldemo.s3.eu-west-2.amazonaws.com/image.png?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAJJWZ7B6WCRGMKFGQ%2F20180210%2Feu-west-2%2Fs3%2Faws4_request&X-Amz-Date=20180210T171315Z&X-Amz-Expires=1800&X-Amz-Signature=12b74b0788aa036bc7c3d03b3f20c61f1f91cc9ad8873e3314255dc479a25351&X-Amz-SignedHeaders=host
```

Depending on your API implementation it might expect a different format. A good approach would be to test this code with your specific API.